xbmc/interfaces/json-rpc/AudioLibrary.cpp

```diff
@@ -51,16 +51,28 @@ JSONRPC_STATUS CAudioLibrary::GetArtists(const std::string &method, ITransportLa
   if (!musicUrl.FromString("musicdb://artists/"))
     return InternalError;
 
+  bool allroles = false;
+  if (parameterObject["allroles"].isBoolean())
+    allroles = parameterObject["allroles"].asBoolean();
+  
   int genreID = -1, albumID = -1, songID = -1;
   const CVariant &filter = parameterObject["filter"];
-  if (filter.isMember("genreid"))
-    genreID = (int)filter["genreid"].asInteger();
-  else if (filter.isMember("genre"))
-    musicUrl.AddOption("genre", filter["genre"].asString());
+  
+  if (allroles)
+    musicUrl.AddOption("roleid", -1000); //All roles, override implicit roleid=1 filter required for backward compatibility
```

Comment:
Any negative number will do. It is related to having to implicitly filter by role=1 for backward compatibility.I hoped that the comment in the code explained sufficiently